### ISSUE TITLE
Models in package Buildings.Occupants.BaseClasses.Validation fail verefication due to differencies in random number generation

### ISSUE THREAD

**Issue by AndreaBartolini** [OpenModelica/OpenModelica#12725]
The following models in the package `Buildings.Occupants.BaseClasses.Validation` fail the verification vs the Dymola reference files:
- `BinaryVariableGeneration`
- `ExponentialVariableGeneration`
- `Linear1D`
- `Logit1D`
- `Logit1DQuadratic`
- `Logit2D`
- `Weibull1DON`
- `WeibullVariableGeneration`

The common reason of these failures is that the function used to generate the random number
```modelica
  state := Modelica.Math.Random.Generators.Xorshift1024star.initialState(localSeed, globalSeed);
  (r, state) := Modelica.Math.Random.Generators.Xorshift1024star.random(state);

```
sometimes returns different values if used in OM or Dymola: 

![image](https://github.com/user-attachments/assets/148506ce-e422-454d-983a-44d694fd93dc)

maybe this is related with the same problem discussed in #12569

**Comment by AndreaBartolini** [comment]
This is also applicable to the following models that use the same random number generator:

- `Buildings.Occupants.Office.Blinds.Validation.Haldi2008BlindsTIn`
- `Buildings.Occupants.Office.Blinds.Validation.Haldi2008BlindsTOut`
- `Buildings.Occupants.Office.Blinds.Validation.Inkarojrit2008BlindsSolarIntensity`
- `Buildings.Occupants.Office.Windows.Validation.Haldi2008WindowsTIn`
- `Buildings.Occupants.Office.Windows.Validation.Haldi2008WindowsTOut`
- `Buildings.Occupants.Office.Windows.Validation.Herkel2008WindowsCompletelyOpen`
- `Buildings.Occupants.Office.Windows.Validation.Herkel2008WindowsSmall`
- `Buildings.Occupants.Office.Windows.Validation.Herkel2008WindowsTiltedOpen`
- `Buildings.Occupants.Office.Windows.Validation.Yun2008WindowsTIn`
- `Buildings.Occupants.Office.Windows.Validation.Yun2008WindowsTOut`
- `Buildings.Occupants.Office.Windows.Validation.Zhang2012WindowsAll`
- `Buildings.Occupants.Office.Windows.Validation.Zhang2012WindowsEast`
- `Buildings.Occupants.Office.Windows.Validation.Zhang2012WindowsNorth`
- `Buildings.Occupants.Office.Windows.Validation.Zhang2012WindowsSouth`
- `Buildings.Occupants.Office.Windows.Validation.Zhang2012WindowsWest`
- `Buildings.Occupants.Residential.Heating.Validation.Nicol2001HeatingEU`
- `Buildings.Occupants.Residential.Heating.Validation.Nicol2001HeatingPakistan`
- `Buildings.Occupants.Residential.Heating.Validation.Nicol2001HeatingUK`
- `Buildings.Occupants.Residential.Windows.Validation.Nicol2001WindowsEU`
- `Buildings.Occupants.Residential.Windows.Validation.Nicol2001WindowsPakistan`
- `Buildings.Occupants.Residential.Windows.Validation.Nicol2001WindowsUK`

**Comment by casella** [comment]
@mwetter, my suggestion if you want to do cross-tool checking is that all the examples using random numbers are set with an explicit seed parameter, rather than relying on the internal mechanisms to generate it automatically. This would solve all the problems. What do you think?

**Comment by mwetter** [comment]
@casella : I agree. This package was contributed by others and should be refactored to store the state in the calling model or block. The implementation should be done correctly by passing the state from one call to the next.

**Comment by casella** [comment]
Shall I open a ticket on the Buildings repo or can you take care of that?
